### DISCUSSION TITLE
V2 + umi + antd3 启动项目构建 dll 文件时出错。

### DISCUSSION THREAD

**Discussion by OKAERI** [alibaba/formily#1301] [Q&A 2.x]
按照官网上进行一一安装
npm install --save @formily/core
npm install --save @formily/react
npm install --save antd moment @formily/antd

**其他项目依赖信息**

"@formily/antd": "^2.0.0-beta.36",
"@formily/core": "^2.0.0-beta.36",
"@formily/react": "^2.0.0-beta.36",
"antd": "^3.19.5",
"dva": "^2.6.0-beta.6",
"moment": "^2.29.1",
"react": "^16.8.6",
"react-dom": "^16.8.6"

**完成的错误提示**

formily yarn start
yarn run v1.21.1
warning ../../../package.json: No license field
$ umi dev
Building dll...

✖ Webpack
  Compiled with some errors in 16.99s

 ERROR  Failed to compile with 3 errors 

This dependency was not found:

* antd/es/space/style/index in ./node_modules/@formily/antd/esm/array-table/style.js, ./node_modules/@formily/antd/esm/editable/style.js and 1 other

To install it, you can run: npm install --save antd/es/space/style/index

 ERROR CODE  ERR_WEBPACK_DLL_MODULE_NOT_FOUND

 DETAILS

## 错误原因

构建 dll 文件时出错了，由于 dependencies 下的依赖会全部被打到 dll 里，所以报错可能有以下原因：

    1. dependencies 里的某些依赖没有 main 文件
    2. 应该放在 devDependencies 里的 node 相关的依赖却写在了 dependencies

## 解决方案

让不该打到 dll 的不被打到 dll 里，有几个解决方案：

    1. 把 node 相关依赖放到 devDependencies 里
    2. 把出错的依赖配置到 dll 的 exclude 里

比如：

    dll: {
      exclude: ['foo']
    }

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by janryWang** [comment]
能看看具体是哪个dependencies没有main文件吗？

**Reply by OKAERI** [reply]
我在dependencies依赖中把 **"@formily/antd": "^2.0.0-beta.36",** 删除掉就可以启动项目了

**Reply by janryWang** [reply]
@formily/antd这个包指定了main呀

**Reply by OKAERI** [reply]
但是如果项目开启DLL的话就不行了，只能关闭dll，然后就都可以是用了，谢谢啦~！

**Reply by janryWang** [reply]
那需要formily做什么能解决这个问题呢
